File: packages/appwsgi/415.tar.bz2/trunk/www/appointment/appointment.js
// Copyright(c) gert.cuykens@gmail.com
appointment=
{
 'name':function()
 {
  http.fml=function(xml)
  {
   document.getElementById('cappointments').innerHTML=''
   appointments=
   {
    'id':'appointments',
    'onselect':function()
    {
     document.getElementById('uid').value=appointments.selection[1]
     appointment.ca()
    }
   }
   dg.init(appointments,xml)
  }
  http.xml ='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>name</cmd>\n'
  http.xml+=' <name>'+document.getElementById('appointment').value+'</name>\n'
  http.xml+='</root>'
  http.send()
 },

 'ca':function()
 {
  http.fml=function(xml)
  {
   document.getElementById('cappointments').innerHTML=''
   appointments=
   {
    'id':'appointments',
    'onselect':function()
    {
     document.getElementById('uid').value=appointments.selection[2]
     document.getElementById('aid').value=appointments.selection[3]
     document.getElementById('calendar').value=appointments.selection[4]
     document.getElementById('appointment').value=appointments.selection[5]
     if(appointments.selection[4]=='None')document.getElementById('calendar').value='0000-00-00 00:00:00'
     calendar.display()
    }
   }
   dg.init(appointments,xml)
  }
  http.xml ='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>ca</cmd>\n'
  http.xml+=' <uid>'+document.getElementById('uid').value+'</uid>\n'
  http.xml+=' <calendar>'+document.getElementById('calendar').value+'</calendar>\n'
  http.xml+='</root>'
  http.send()
 },

 'find':function()
 {
  http.fml=function(xml)
  {
   document.getElementById('cappointments').innerHTML=''
   appointments=
   {
    'id':'appointments',
    'onselect':function()
    {
     document.getElementById('uid').value=appointments.selection[2]
     document.getElementById('aid').value=appointments.selection[3]
     document.getElementById('calendar').value=appointments.selection[4]
     document.getElementById('appointment').value=appointments.selection[5]
     if(appointments.selection[4]=='None')document.getElementById('calendar').value='0000-00-00 00:00:00'
     calendar.display()
    }
   }
   dg.init(appointments,xml)
  }
  http.xml ='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>find</cmd>\n'
  http.xml+=' <calendar>'+document.getElementById('calendar').value+'</calendar>\n'
  http.xml+=' <appointment>'+document.getElementById('appointment').value+'</appointment>\n'
  http.xml+='</root>'
  http.send()
 },
 
 'save':function()
 {
  var cmd
  if(document.getElementById('aid').value=='new'){cmd='insert'}else{cmd='update'}
  http.fml=function(xml)
  {
   if(document.getElementById('aid').value=='new')document.getElementById('aid').value=xml.getElementsByTagName('id')[0].childNodes[0].nodeValue
   if(c=document.getElementById('cappointments'))while(c.lastChild)c.removeChild(c.lastChild)
  }
  http.xml ='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>'+cmd+'</cmd>\n'
  http.xml+=' <aid>'+document.getElementById('aid').value+'</aid>\n'
  http.xml+=' <uid>'+document.getElementById('uid').value+'</uid>\n'
  http.xml+=' <calendar>'+document.getElementById('calendar').value+'</calendar>\n'
  http.xml+=' <appointment>'+document.getElementById('appointment').value+'</appointment>\n'
  http.xml+='</root>'
  http.send()
 },

 'remove':function()
 {
  http.fml=function(xml)
  {
   document.getElementById('aid').value='new'
   document.getElementById('appointment').value=''
   if(c=document.getElementById('appointments'))while(c.lastChild)c.removeChild(c.lastChild)
  }
  http.xml ='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>remove</cmd>\n'
  http.xml+=' <aid>'+document.getElementById('aid').value+'</aid>\n'
  http.xml+='</root>'
  if(document.getElementById('aid').value!='new'){http.send()}
 }
}

File: packages/appwsgi/appwsgi%201050.tar.bz2/appwsgi/www/appointment/appointment.py
# Copyright(c) gert.cuykens@gmail.com
from frame import Frame
from session import Session
from db import Db

def application(environ, response):
    cookie = environ.get('HTTP_COOKIE','')
    document = environ['wsgi.input'].read(int(environ.get('CONTENT_LENGTH','0')))
    db = Db()
    v = Frame.read(document)
    s = Session(db,cookie,v['gid'])

    if v['cmd'] == 'overview':
                               db.execute("SELECT calendar,appointment FROM appointments WHERE calendar >= ? AND calendar < ?",(v['from'],v['to']))
    elif v['cmd'] == 'insert':      
        if s.GID  == 'guest' : db.execute("INSERT INTO appointments VALUES (?,NULL,?,?)",(s.UID,v['calendar'],v['appointment']))
        elif s.GID == 'admin': db.execute("INSERT INTO appointments VALUES (?,NULL,?,?)",(v['uid'],v['calendar'],v['appointment']))
    elif v['cmd'] == 'update':
        if s.GID  == 'guest':  db.execute("UPDATE appointments SET calendar=?, appointment=? WHERE aid=? AND uid=?",(v['calendar'],v['appointment'],v['aid'],s.UID))
        elif s.GID == 'admin': db.execute("UPDATE appointments SET uid=?, calendar=?, appointment=? WHERE aid=?",(v['uid'],v['calendar'],v['appointment'],v['aid']))
    elif v['cmd'] == 'remove':
        if s.GID  == 'guest':  db.execute("DELETE FROM appointments WHERE aid=? AND uid=?",(v['aid'],s.UID))
        elif s.GID == 'admin': db.execute("DELETE FROM appointments WHERE aid=?",(v['aid'],))
    elif v['cmd'] == 'name':
        if s.GID  == 'guest':  db.execute("SELECT uid,name,adress,city,country,phone FROM users WHERE name LIKE ? AND uid=?",('%'+v['name']+'%',s.UID))
        elif s.GID == 'admin': db.execute("SELECT uid,name,adress,city,country,phone FROM users WHERE name LIKE ?",("%"+v['name']+"%",))
    elif v['cmd'] == 'find':
                               db.execute("SELECT users.name, appointments.* FROM users, appointments WHERE users.uid = appointments.uid AND appointments.calendar >= ? AND appointments.appointment LIKE ? GROUP BY appointments.aid",(v['calendar'],"%"+v['appointment']+"%"))
    elif v['cmd'] == 'ca':
                               db.execute("SELECT users.name, appointments.* FROM users, appointments WHERE users.uid = appointments.uid AND appointments.calendar >= ? AND appointments.appointment LIKE ? AND appointments.uid=? GROUP BY appointments.aid",(v['calendar'],"%"+v['appointment']+"%",v['uid']))

    xml = '<?xml version="1.0" encoding="UTF-8"?>\n'
    xml+= '<root>\n'
    xml+= ' <cmd>'+str(v['cmd'])+'</cmd>\n'
    xml+= ' <sid>'+str(s.SID)+'</sid>\n'
    xml+= ' <exp>'+str(s.EXP)+'</exp>\n'
    xml+= ' <uid>'+str(s.UID)+'</uid>\n'
    xml+= ' <gid>'+str(s.GID)+'</gid>\n'
    xml+= Frame.write(db)
    xml+= '</root>'

    response('200 OK', [('Content-type', 'text/xml'), ('Set-Cookie', s.COOKIE)])
    return [xml]

